ids_body.php

```diff
@@ -1,24 +1,37 @@
 <?php
 /*
- * Author Shoichi Chou ( shoichi.chou@gmail.com )
-*/
+ * @author Shoichi Chou ( shoichi.chou@gmail.com )
+ */
 
 // 0614-2016 change server to https://tools.wmflabs.org/idsgen/
 class IDS {
 	static function onParserInit( Parser $parser ) {
 		$parser->setHook( 'ids', [ __CLASS__, 'idsRender' ] );
 		return true;
 	}
+	/*
+	 * This method handles IDS tags.
+	 *
+	 * <ids>⿰電心</ids> ->
+	 * 	<img align=middle class="ids-char" src="https://tools.wmflabs.org/idsgen/⿰電心.svg?字體=宋體" ...>
+	 *
+	 * <ids font=楷體粗體>⿰電心</ids> ->
+	 * 	<img align=middle class="ids-char" src="https://tools.wmflabs.org/idsgen/⿰電心.svg?字體=楷體粗體" ...>
+	 *
+	 * @see https://www.mediawiki.org/wiki/Manual:Tag_extensions
+	 */
 	static function idsRender( $input, array $args, Parser $parser, PPFrame $frame ) {
-	  /*
-	  Example:
-	  $ret ='<img alt="⿰電心" src="https://tools.wmflabs.org/idsgen/⿰電心.png?字體=宋體" width="16" height="16"  />';
-	  */
-	  $ret = '<img  align=middle alt="';
-	  $ret .=$input;
-	  $ret .= '" src="https://tools.wmflabs.org/idsgen/';
-	  $ret .= $input;
-	  $ret .= '.png?字體=宋體" style="height: 1em; width: 1em; vertical-align: middle; margin: 0.4em 0px 0.7em; "  />';
-	  return $ret;
+		// Support for Simplified "體" (font)
+		$font = isset($args['font']) ? strtr($args['font'] , '体', '體') : '宋體';
+		$src = 'https://tools.wmflabs.org/idsgen/' . rawurlencode($input) . '.svg?字體=' . rawurlencode($font);
+
+		return Html::element('img', [
+			'align' => 'middle',
+			'class' => 'ids-char',  // helps with custom styling
```

Comment:
If the user constantly faces PNG fallbacks but still wants crispy edges, they can use:

```CSS
img.ids_char {
    image-rendering: -moz-crisp-edges;  /* Firefox */ 
    image-rendering: -o-crisp-edges;  /* Opera */ 
    image-rendering: -webkit-optimize-contrast;  /* Webkit (non-standard naming) */ 
    image-rendering: crisp-edges;
    -ms-interpolation-mode: nearest-neighbor;  /* IE (non-standard property) */ 
}
```